cmd/entrypoint/cmd/root.go

```diff
@@ -0,0 +1,202 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	constants "github.com/cybozu-go/github-actions-controller"
+	"github.com/cybozu-go/github-actions-controller/agent"
+	"github.com/cybozu-go/well"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	// Environments
+	podName           = os.Getenv(constants.PodNameEnvName)
+	podNamespace      = os.Getenv(constants.PodNamespaceEnvName)
+	runnerToken       = os.Getenv(constants.RunnerTokenEnvName)
+	runnerOrg         = os.Getenv(constants.RunnerOrgEnvName)
+	runnerRepo        = os.Getenv(constants.RunnerRepoEnvName)
+	extendDuration    = os.Getenv(constants.ExtendDurationEnvName)
+	slackAgentSvcName = os.Getenv(constants.SlackAgentServiceNameEnvName)
+
+	// Directory/File Paths
+	runnerDir = filepath.Join("/runner")
+	workDir   = filepath.Join(runnerDir, "_work")
+
+	extendFlagFile    = filepath.Join(os.TempDir(), "extend")
+	failureFlagFile   = filepath.Join(os.TempDir(), "failure")
+	cancelledFlagFile = filepath.Join(os.TempDir(), "cancelled")
+	successFlagFile   = filepath.Join(os.TempDir(), "success")
+
+	configCommand = filepath.Join(runnerDir, "config.sh")
+	runSVCCommand = filepath.Join(runnerDir, "bin", "runsvc.sh")
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "entrypoint",
+	Short: "GitHub Actions runner Entrypoint",
+	Long:  "GitHub Actions runner Entrypoint",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		return checkEnvs()
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := os.Mkdir(workDir, 0777); err != nil {
```

Comment:
Is `0777` correct permission?

Reply:
Indeed, the `0777` permissions may have been strong.
However, I wasn't sure what the appropriate permissions were, so I set them to `0755`.